### PR TITLE
[Tracker Alignment] Miscellanea graphical fixes for PV-related validations

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Follow-up of PR https://github.com/cms-sw/cmssw/pull/36178, fixes an off-by-one bin mistake in the population of the Split Vertex resolution plots.
I profit to fix a couple of other bug in the `FitPVResiduals.C` macro.

#### PR validation:

Private. Tested on 2021 Beam Test data, see e.g. [here](http://musich.web.cern.ch/musich/testBeamTest/splitVertex/v5/)
image

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport, but will need to be backported together with #36178

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/OfflineValidation/src/PVValidationHelpers.cc [hunk 1]
```diff
@@ -165,9 +165,11 @@ std::vector<float> PVValHelper::generateBins(int n, float start, float range)
...
   std::iota(v.begin(), v.end(), 1.);
 
-  //std::cout<<" interval:"<<interval<<std::endl;
-  //for(float &a : v) { std::cout<< a << " ";  }
-  //std::cout<< "\n";
+  /*
+    std::cout<<" interval:"<<interval<<std::endl;
+    for(float &a : v) { std::cout<< a << " ";  }
+    std::cout<< "\n";
+  */
```
Comment (tvami) [inline Alignment/OfflineValidation/src/PVValidationHelpers.cc:172]: Can this be removed?

Comment (mmusich) [reply Alignment/OfflineValidation/src/PVValidationHelpers.cc:172]: no.

**Comment by tvami** [comment]
+alca
 * tests pass

File: Alignment/OfflineValidation/macros/FitPVResolution.C [hunk 2]
```diff
@@ -335,17 +335,17 @@ void FitPVResolution(TString namesandlabels, TString theDate = "", bool isStrict
   }
 
   // max vertices
-  const int max_n_vertices = std::min(40, VTXBINS);  // take the minimum to avoid overflow
+  const int max_n_vertices = std::min(60, VTXBINS);  // take the minimum to avoid overflow
   std::vector<float> myNVtx_bins_;
   for (float i = 0; i <= max_n_vertices; i++) {
```
Comment (perrotta) [inline Alignment/OfflineValidation/macros/FitPVResolution.C:340]:
```suggestion
  for (float i = 0; i < max_n_vertices + 0.5f ; i++) {
```
(I wouldn't trust a check of equalities between float's...)

File: Alignment/OfflineValidation/macros/FitPVResolution.C [hunk 3]
```diff
@@ -335,17 +335,17 @@ void FitPVResolution(TString namesandlabels, TString theDate = "", bool isStrict
...
-    myNVtx_bins_.push_back(1. + i);
+    myNVtx_bins_.push_back(i - 0.5f);
   }
 
   // max track
-  const int max_n_tracks = std::min(60, TRACKBINS);  // take the minimum to avoid overflow
+  const int max_n_tracks = std::min(120, TRACKBINS);  // take the minimum to avoid overflow
   std::vector<float> myNTrack_bins_;
   for (float i = 0; i <= max_n_tracks; i++) {
```
Comment (perrotta) [inline Alignment/OfflineValidation/macros/FitPVResolution.C:347]: ditto

File: Alignment/OfflineValidation/plugins/SplitVertexResolution.cc [hunk 4]
```diff
@@ -295,12 +296,55 @@ SplitVertexResolution::SplitVertexResolution(const edm::ParameterSet& iConfig)
 
   mypT_bins_ = PVValHelper::makeLogBins<float, nPtBins_>(sumpTStartScale_, sumpTEndScale_);
 
-  std::vector<float> vect = PVValHelper::generateBins(nTrackBins_ + 1, 1., 120.);
+  // IMPORTANT
+  // first argument is start, second argument is the range so it's [-0.5;nTracksBins_-0.5]
```
Comment (perrotta) [inline Alignment/OfflineValidation/plugins/SplitVertexResolution.cc:300]: There are three arguments in fact: I guess the comment can be made more clear...

File: Alignment/OfflineValidation/plugins/SplitVertexResolution.cc [hunk 5]
```diff
@@ -295,12 +296,55 @@ SplitVertexResolution::SplitVertexResolution(const edm::ParameterSet& iConfig)
...
+  std::vector<float> vect = PVValHelper::generateBins(nTrackBins_ + 1, -0.5, nTrackBins_);
   std::copy(vect.begin(), vect.begin() + nTrackBins_ + 1, myNTrack_bins_.begin());
 
   vect.clear();
-  vect = PVValHelper::generateBins(nVtxBins_ + 1, 1., 40.);
+
+  // IMPORTANT
+  // first argument is start, second argument is the range so it's [-0.5;nVtxBins_-0.5]
```
Comment (perrotta) [inline Alignment/OfflineValidation/plugins/SplitVertexResolution.cc:307]: ditto

**Review by perrotta — COMMENTED** [review]
@mmusich take the following as comments that you can take into account if you plan to further update/fix the code.

**Comment by perrotta** [comment]
+1

**Comment by aandvalenzuela** [comment]
Hello @mmusich,

Could it be that changes in this PR broke Unit Test `testPVPlotting` (`Aligment/OfflineValidation`) on `el8_ppc64le_gcc11`. It has been failing in IBs since a few days (https://cmssdt.cern.ch/SDT/cgi-bin/logreader/el8_ppc64le_gcc11/CMSSW_13_0_X_2022-12-12-2300/unitTestLogs/Alignment/OfflineValidation#/15100-15100).

Thank you,
Andrea.

**Comment by mmusich** [comment]
@aandvalenzuela this PR is more than 1 year old....

**Comment by aandvalenzuela** [comment]
Ups, sorry I completely overlooked the year. I will open a new issue for this failure

**Pull request merged**